### ISSUE TITLE
doc task does not regenerate scaladocs when api directory is removed

### ISSUE THREAD

**Issue by jaburjak** [sbt/sbt#7078]
## steps

With sbt 1.7.1:

1. generate scaladocs by running `sbt doc`
2. remove directory `target/scala-3.2.0/api`
3. run `sbt doc` again

## problem

Running `sbt doc` after removing the `api` directory does not recreate it. sbt reports success instead:

```
$ sbt doc
downloading sbt launcher 1.8.0
[info] welcome to sbt 1.7.1 (Amazon.com Inc. Java 11.0.17)
[info] loading settings for project scalapr-build from plugins.sbt ...
[info] loading project definition from /builds/jaburjak/scalapr/project
[info] loading settings for project root from build.sbt ...
[info] set current project to scalapr (in build file:/builds/jaburjak/scalapr/)
[success] Total time: 4 s, completed Nov 15, 2022, 11:23:09 PM
```

## expectation

sbt should notice that the `api` directory was removed and recreate it.

## notes

Creating empty `target/scala-3.2.0/api` directory after removing it makes sbt notice that scaladocs are missing and it will recreate them.

I stumbled upon this issue when setting up continuous integration pipeline for my project. To speed up builds, I enabled caching of the `target` directory between pipelines. The last stage of the pipeline is to generate scaladocs and publish them to GitLab Pages, which it does by running `sbt doc` and then moving `target/scala-3.2.0/api` to another directory that gets published by Pages.

This worked the first time the pipeline was run. Then on subsequent runs, because the `target` directory was cached (I assume there’s something in there telling sbt that scaladocs are already generated) but `api` directory was not there (cache is created at the end of the pipeline), the pipeline broke because sbt did not recreate it.

**Comment by atoz96** [comment]
Hi, @eed3si9n 
Can I work on this issue?

**Comment by eed3si9n** [comment]
@atoz96 Yes, please. As per usual, please check out our [Contributor's Guildeline](https://github.com/sbt/sbt/blob/develop/CONTRIBUTING.md) and:
1. Make sure you can reproduce the problem and the fix manually
2. Include tests for the change
3. Follow our AI assisted contributions guideline (please disclose AI usages in the PR description, if any)
4. Multiple contributors might work on this issue in parallel
5. Generally we recommend getting CI to pass on a forked repo before sending a pull request (pull requests that do not pass tests will be closed after a few days)

A good starting point might be to check if this is still reproducible using sbt 2.0.0-RC9.
